src/Filter/ModelFilter.php

```diff
@@ -140,4 +139,37 @@ private function getParentAlias(ProxyQueryInterface $query, string $alias): stri
 
         return $parentAlias;
     }
+
+    private function buildMultipleAnd(ProxyQueryInterface $query, string $alias, FilterData $data): Orx
```

Comment:
This function will now do the magic....

For each entity that is added to the filter, it finds its identifiers, and for each identifier it adds a condition to the query, if the identifier value is an object and the type is known, is added to the setParameter also.

This works in all scenarios.

* Single identifiers as integers
* Single identifiers as object (uuid)
* Compound identifiers

Reply:
I need to add tests for this. For now I have tested it with the entities provided in the test folder.

Reply:
What I like less is having to replicate this same logic for the autocomplete type, wdyt @VincentLanglet ?

Reply:
The ModelAutocomplete Filter is deprecated

Reply:
Then we can simply not to the change there. I will add a test for the autocomplete Type using the modelFilter, so we are covered.